docs/pages/deploy-a-cluster/multi-region-blueprint.mdx

```diff
@@ -0,0 +1,222 @@
+
+
+### Deploying Teleport
+
+Once all Teleport dependencies are set up, you can deploy Teleport via the
+`teleport-cluster` Helm chart. You need to create one release per Kubernetes cluster.
+The values for a specific region look like:
+
+```yaml
+chartMode: standalone
+clusterName: teleport-multi-region.example.org
+persistence:
+  enabled: false
+enterprise: true
+
+auth:
+  teleportConfig:
+    # Configure CockroachDB
+    teleport:
+      storage:
+        type: cockroachdb
+        # regional CockraochDB URL, if you followed the Cockroach docs it looks like
+        # cockroachdb-public.<region>.svc.cluster.local
+        conn_string: postgres://teleport@cockroachdb-public.<region>.svc.cluster.local:26257/teleport_backend?sslmode=verify-full&pool_max_conns=20
+        audit_events_uri:
+          - "postgres://teleport@cockroachdb-public.<region>.svc.cluster.local:26257/teleport_audit?sslmode=verify-full"
+        # replace this with the URI of your regional object storage (S3, GCS, MinIO)
+        audit_sessions_uri: "uri://to-your-regional-bucket"
+        ttl_job_cron: '*/20 * * * *'
+    
+    # Configure proxy peering
+    auth_service:
+      tunnel_strategy:
+        type: proxy_peering
+        agent_connection_count: 2
+
+  # Mount the CockroachDB certs and have Teleport use them (via default env vars)
+  extraVolumes:
+    - name: db-certs
+      secret:
+        secretName: cockroach-certs
+  extraVolumeMounts:
+    - name: db-certs
+      mountPath: /var/lib/db-certs
+      readOnly: true
+  extraEnv:
+    - name: PGSSLROOTCERT
+      value: /var/lib/db-certs/ca.crt
+    - name: PGSSLCERT
+      value: /var/lib/db-certs/client.teleport.crt
+    - name: PGSSLKEY
+      value: /var/lib/db-certs/client.teleport.key
+
+# Pass the TLS certificate use by the proxy
+# This can be omitted if you use cert-manager and set `highAvailability.certManager`
+tls:
+  existingSecretName: proxy-cert
+
+highAvailability:
+  replicaCount: 2
+```
+
+### GeoDNS Setup
+
+Finalize the deployment by setting up GeoDNS to route the users to the closest
+Teleport deployment. The setup will differ from one DNS provider to another. Use
+short TTLs to ensure you can quickly steer traffic off a region if needed.
+
+When possible, configure healthchecks on `/v1/webapi/ping` so a broken/faulty
+Teleport region is automatically removed from the DNS records and users are
+routed to the closest working Teleport deployment.
+
+## Update strategy
+
+Updating Teleport on multi-region setups poses several challenges.
+Strictly following Teleport compatibility guarantees requires downscaling to a
+single auth before updating. In a multi region setup, this requires:
+
+- relying on a single region for auth during the udpate
+- suspending two regions, ot allowing proxies to connect to auths from a
+  different region (this is not the case by default and requires a Cilium
+  cross-cluster service and a custom join token for the Proxy Service).
+
+<Admonition type="note">
```

Comment:
I'd be more prescriptive here and say:

You can perform rolling upgrades, when there are multiple auth servers of different versions running at the same time.

For example, let's assume there are multiple versions of Teleport available:

```
14.5.3
14.5.4
14.5.6
14.6.1
14.6.2
15.0.0
15.0.1
```

* You upgrade between **any** patch versions of the same minor and major release, e.g. `14.5.3` to `14.5.6`
* You upgrade between successive minor versions, e.g. `14.5.3` to `14.6.1`
* You upgrade between strictly previous minor version and next major version, for example `14.6.1` to `15.0.1`.

In all other cases, the results of a rolling upgrade may result in data corruption and outage.

Reply:
I tried to come up with such rules but according to @rosstimothy we can lose data even by rolling update between two patches. [Slack link](https://gravitational.slack.com/archives/C03PJTF5BPD/p1717014431273299?thread_ts=1715189733.223619&cid=C03PJTF5BPD). We're just accepting the risk in cloud as we're following the version very closely and the probability is low, but the risk is always non-null.